### PR TITLE
Owlv2

### PR DESCRIPTION
**Pull request by probicheaux**
# Description

Add owlv2 to inference

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

staging with ui

## Any specific deployment considerations

inference-internal updated

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

**Review by SolomonLake — DISMISSED** [review]
🚢

**Review by SolomonLake — APPROVED** [review]

**Comment by probicheaux** [comment]
@PawelPeczek-Roboflow this is ready for review

File: inference/core/entities/requests/owlv2.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    cls: str
+
+
+class TrainingImage(BaseModel):
+    boxes: List[TrainBox]
+    image: InferenceRequestImage
+
+
+class OwlV2InferenceRequest(BaseRequest):
+    """Request for gaze detection inference.
```
Comment (PawelPeczek-Roboflow) [inline inference/core/entities/requests/owlv2.py:27]: Please use `Field(...)` with descriptions and defaults placed there, that is to push the info into OpenAPI docs

File: inference/models/owlv2/owlv2.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+                "y": float(y),
+                "w": float(w),
+                "h": float(h),
+                "confidence": float(score),
+            }
+            for c, (x, y, w, h), score in zip(pred_classes, pred_boxes, pred_scores)
+        ]
+
+    def infer(self, image, training_data, confidence=0.99, **kwargs):
+        print("UNUSED KWARGS")
```
Comment (PawelPeczek-Roboflow) [inline inference/models/owlv2/owlv2.py:215]: remove print

File: tests/inference/models_predictions_tests/test_owlv2.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from inference.core.entities.requests.owlv2 import OwlV2InferenceRequest
```
Comment (PawelPeczek-Roboflow) [inline tests/inference/models_predictions_tests/test_owlv2.py:1]: make it pytest test

**Review by PawelPeczek-Roboflow — COMMENTED** [review]
LGTM, besides minor comments and please add at least minimalistic docs describing what the model is useful for and how to make requests.

regarding `CORE_MODEL_OWLV2_ENABLED=True` in env - shall I assume the intent is to expose the model behind lambda?

**Review by PawelPeczek-Roboflow — APPROVED** [review]

**Pull request merged**